Boltdir/site-modules/metrics/spec/files/collect_metrics_files_spec.rb

```diff
@@ -1,5 +1,241 @@
+    it "Correctly filters out subdirs known to have no data and continues on with the rest" do
+      expect(CMF_obj).not_to receive(:inspect_service_dir_for_metrics_files)
+        .with("#{puppet_metrics_collector_fixture}/bin", "bin")
+      expect(CMF_obj).not_to receive(:inspect_service_dir_for_metrics_files)
+        .with("#{puppet_metrics_collector_fixture}/scripts", "scripts")
+      expect(CMF_obj).to receive(:inspect_service_dir_for_metrics_files)
+        .with("#{puppet_metrics_collector_fixture}/puppetdb", "puppetdb")
+      expect(CMF_obj).to receive(:inspect_service_dir_for_metrics_files)
+        .with("#{puppet_metrics_collector_fixture}/puppetserver", "puppetserver")
+      expect(CMF_obj).to receive(:inspect_service_dir_for_metrics_files)
+        .with("#{puppet_metrics_collector_fixture}/orchestrator", "orchestrator")
+      CMF_obj.inspect_metrics_dir_for_service_dirs
+    end
+  end
+
+  describe "#inspect_service_dir_for_metrics_files" do
+    it "Calls methods to look at the tar and json files in the fixture" do
+      expect(CMF_obj).to receive(:consider_tarfile_contents_for_inclusion)
+        .with("#{puppet_metrics_collector_fixture}/puppetdb/puppetdb-2019.07.09.02.45.01.tar.gz", "puppetdb")
+      expect(CMF_obj).to receive(:consider_tarfile_contents_for_inclusion)
+        .with("#{puppet_metrics_collector_fixture}/puppetdb/puppetdb-2019.07.17.02.45.01.tar.gz", "puppetdb")
+      expect(CMF_obj).to receive(:consider_json_for_inclusion)
+        .with("#{puppet_metrics_collector_fixture}/puppetdb/ip-10-227-1-138.amz-dev.puppet.net/20190718T092001Z.json")
+      expect(CMF_obj).to receive(:consider_json_for_inclusion)
+        .with("#{puppet_metrics_collector_fixture}/puppetdb/ip-10-227-1-138.amz-dev.puppet.net/20190718T173001Z.json")
+      CMF_obj.inspect_service_dir_for_metrics_files("#{puppet_metrics_collector_fixture}/puppetdb",
+                                                    "puppetdb")
+    end
+  end
+
+  describe "#json_file_from_target_window?" do
+    it "returns false for a json from 30h before the window" do
+      expect(CMF_obj.json_file_from_target_window?(json_name_30h_before_window)).to eq(false)
+    end
+    it "returns false for a json from 5h before the window" do
+      expect(CMF_obj.json_file_from_target_window?(json_name_5h_before_window)).to eq(false)
+    end
+    it "returns true for a json from inside the window" do
+      expect(CMF_obj.json_file_from_target_window?(json_name_middle_of_window)).to eq(true)
+    end
+    it "returns false for a json from 5h after the window" do
+      expect(CMF_obj.json_file_from_target_window?(json_name_5h_after_window)).to eq(false)
+    end
+    it "returns false for a json from 30h after the window" do
+      expect(CMF_obj.json_file_from_target_window?(json_name_30h_after_window)).to eq(false)
+    end
+  end
+
+  describe "#consider_json_for_inclusion?" do
+    it "calls stage_json when json_file_from_target_window? returns true" do
```

Comment:
It would be helpful to use contexts to make the test cases more clear / readable (as in RAS).

Reply:
great catch... I knew things felt wrong, but couldn't put my finger on it.